src/neighbors.cc

```diff
@@ -1237,24 +1238,32 @@ peerDNSConfigure(const ipcache_addrs *ia, const Dns::LookupDetails &, void *data
 }
 
 static void
-peerRefreshDNS(void *data)
+peerScheduleDnsRefreshCheck(const double delayInSeconds)
 {
-    CachePeer *p = nullptr;
-
-    if (eventFind(peerRefreshDNS, nullptr))
-        eventDelete(peerRefreshDNS, nullptr);
+    if (eventFind(peerDnsRefreshCheck, nullptr))
+        eventDelete(peerDnsRefreshCheck, nullptr);
+    eventAddIsh("peerDnsRefreshCheck", peerDnsRefreshCheck, nullptr, delayInSeconds, 1);
+}
 
-    if (!data && 0 == stat5minClientRequests()) {
+static void
+peerDnsRefreshCheck(void *)
+{
+    if (!stat5minClientRequests()) {
         /* no recent client traffic, wait a bit */
-        eventAddIsh("peerRefreshDNS", peerRefreshDNS, nullptr, 180.0, 1);
+        peerScheduleDnsRefreshCheck(180.0);
         return;
     }
 
-    for (p = Config.peers; p; p = p->next)
-        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p);
+    peerDnsRefreshStart();
+}
+
+static void
+peerDnsRefreshStart()
+{
+    for (const auto &p: CurrentCachePeers())
+        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p.get());
```

Comment:
This PR incorrectly attributes this loop change to PR 1950. That PR (i.e. master/v7 commit https://github.com/squid-cache/squid/commit/dd3292c) does not change this loop. To backport that commit (without also backporting much more invasive commit https://github.com/squid-cache/squid/commit/2e24d0bf changes), this loop has to be moved like this AFAICT:


```suggestion
    for (auto p = Config.peers; p; p = p->next)
        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p);
```

The suggested change should fix `CurrentCachePeers was not declared in this scope` build errors in this PR.